extensions/amp-smartlinks/0.1/amp-smartlinks.js

```diff
@@ -0,0 +1,218 @@
+            this.runSmartlinks_();
+          });
+        });
+  }
+
+  /**
+   * Wait for the config promise to resolve and then proceed to functionality
+   * @private
+   */
+  runSmartlinks_() {
+    this.getLinkmateOptions_().then(config => {
+      this.linkmateOptions_.linkmateExpected = config['linkmate_enabled'];
+      this.linkmateOptions_.publisherID = config['publisher_id'];
+
+      this.postPageImpression_();
+      this.linkmate_ = new Linkmate(
+          /** @type {!../../../src/service/ampdoc-impl.AmpDoc} */
+          (this.ampDoc_),
+          /** @type {!../../../src/service/xhr-impl.Xhr} */
+          (this.xhr_),
+          /** @type {!Object} */
+          (this.linkmateOptions_)
+      );
+      this.smartLinkRewriter_ = this.initLinkRewriter_();
+
+      // If the config specified linkmate to run and our API is expecting
+      // linkmate to run
+      if (this.linkmateOptions_.linkmateEnabled &&
+          this.linkmateOptions_.linkmateExpected) {
+        this.smartLinkRewriter_.getAnchorReplacementList();
+      }
+    });
+  }
+
+  /**
+   * API call to retrieve the Narrativ config for this extension.
+   * API response will be a list containing nested json values. For the purpose
+   * of this extension there will only ever be one value in the list:
+   *  {amp_config: {linkmate_enabled: <!boolean>, publisher_id: <!number>}}
+   * @return {?Promise<!JsonObject>}
+   * @private
+   */
+  getLinkmateOptions_() {
+    const fetchUrl = ENDPOINTS.NRTV_CONFIG_ENDPOINT.replace(
+        '.nrtv_slug.', this.linkmateOptions_.nrtvSlug
+    );
+
+    try {
+      return this.xhr_.fetchJson(fetchUrl, {
+        method: 'GET',
+        ampCors: false,
+      })
+          .then(res => res.json())
+          .then(res => {
+            return getData(res)[0]['amp_config'];
+          });
+    } catch (err) {
+      return null;
+    }
+  }
+
+
+  /**
+   * API call to indicate a page load event happened
+   * @private
+   */
+  postPageImpression_() {
+    // When using layout='nodisplay' manually trigger CustomEventReporterBuilder
+    this.signals().signal(CommonSignals.LOAD_START);
+    const payload = this.buildPageImpressionPayload_();
+
+    const builder = new CustomEventReporterBuilder(this.element);
+
+    builder.track('page-impression', ENDPOINTS.PAGE_IMPRESSION_ENDPOINT);
+
+    builder.setTransportConfig(dict({
+      'beacon': true,
+      'image': false,
+      'xhrpost': true,
+      'useBody': true,
+    }));
+
+    builder.setExtraUrlParams(payload);
+    const reporter = builder.build();
+
+    reporter.trigger('page-impression');
+  }
+
+  /**
+   * Initialize and register a Narrativ LinkRewriter instance
+   * @return {!../../amp-skimlinks/0.1/link-rewriter/link-rewriter.LinkRewriter}
+   * @private
+   */
+  initLinkRewriter_() {
+    const options = {linkSelector: this.linkmateOptions_.linkSelector};
+
+    return this.linkRewriterService_.registerLinkRewriter(
+        SMARTLINKS_REWRITER_ID,
```

Comment:
nit remove from constants.js and export a const named TAG in this class as we do in other extensions and use this in all places where 'amp-smartlinks' string is used. 

    export TAG = 'amp-smartlinks';

You can address in a separate PR.

Reply:
Makes sense. I added it here as it's a small change.